core/dbt/task/show.py

```diff
@@ -23,13 +26,19 @@ def execute(self, compiled_node, manifest):
         # Allow passing in -1 (or any negative number) to get all rows
         limit = None if self.config.args.limit < 0 else self.config.args.limit
 
+        compiled_code = compiled_node.compiled_code
+
         if "sql_header" in compiled_node.unrendered_config:
-            compiled_node.compiled_code = (
```

Comment:
Previously we were mutating the `compiled_code` attribute on `compiled_node` directly in addition to executing it, whereas the changes here just execute it.

I imagine modifying `compiled_node.compiled_code` had some downstream effects such as surfacing the `sql_header` in the `target/compiled` files. Is the motivation for this change to be more consistent with what target/compiled files typically contain for run models?

Reply:
> Is the motivation for this change to be more consistent with what target/compiled files typically contain for run models?

Yes, this is what I was thinking. The `sql_header` isn't part of a node's `compiled_code` when it's compiled or materialized, so it feels inconsistent that it would be included just for the manifest produced by `show`.